bin/npm-cli.js

```diff
@@ -75,6 +75,7 @@
   npm.load(conf, function (er) {
     if (er) return errorHandler(er)
     if (
+      !npm.argv.includes('npm') &&
```

Comment:
You'll note that [`pnpm` also makes sure it's a relevant command, as well as whether we're in `global` mode](https://github.com/pnpm/pnpm/blob/31595a918f61e1be16d225a260941f5c1b7c02e6/packages/pnpm/src/main.ts#L170). I think both of those check are worth doing.

I also agree with what pnpm is doing as far as disabling this on CI. The [`ci-info` package by @watson](https://www.npmjs.com/package/ci-info) seems to do a pretty good job at detecting this, so maybe that's worth pulling in (/cc @iarna).

Reply:
Thanks @zkat! I made updates to check install/update and if global mode.

Disregarding the CI stuff here based on comments on #33